runtime/lua/vim/fcnotify.lua

```diff
@@ -0,0 +1,224 @@
+-- Create a file watcher, each watcher is identified by the name of the file that it
+-- watches.
+
+local uv = vim.loop
+local Watcher = {}
+Watcher.__index = Watcher
+local WatcherList = {}
+local check_handle = nil
+
+-- Checks if a buffer should have a watcher attached to it.
+local function valid_buf(bufnr)
+  if not vim.api.nvim_buf_is_valid(bufnr) then
+    return false
+  end
+
+  local buflisted = vim.api.nvim_buf_get_option(bufnr, 'buflisted')
+  local buftype = vim.api.nvim_buf_get_option(bufnr, 'buftype')
+
+  return buflisted or buftype == '' or buftype == 'acwrite'
+end
+
+-- Callback for the check handle, checks if there are pending notifications
+-- for any watcher, and handles them as per the value of the `fcnotify`
+-- option.
+local function check_notifications()
+  for _, watcher in pairs(WatcherList) do
+    if watcher.pending_notifs == true and valid_buf(watcher.bufnr) then
```

Comment:
don't need to check `== true`, just check `if watcher.pending_notifs and valid_buf(watcher.bufnr)`